### PR TITLE
Add citation information to README

### PR DESCRIPTION
**Pull request by AustinT**
1. Add short table of papers that have used syntheseus so far
2. Add bibtex for Faraday discussion paper to help people cite paper in the future

### CODE REVIEW

**Review by mrwnmsr — APPROVED** [review]

File: README.md [hunk 1]
```diff
@@ -41,6 +41,31 @@ pip install "syntheseus[all]"
...
+## Citation and usage
+
+Since the release of our package, we've been thrilled to see syntheseus be used in the following projects:
+
+| **Project** | **Usage** | **Reference(s)** |
+|:--------------|:-----|:-----------|
+|Retro-fallback search|Multi-step search|ICLR [paper](https://arxiv.org/abs/2310.09270), [code](https://github.com/AustinT/retro-fallback-iclr24)|
+|RetroGFN|Pre-packaged single-step models|arXiv [paper](https://arxiv.org/abs/2406.18739), [code](https://github.com/gmum/RetroGFN)|
+|TANGO|Single-step and multi-step|arXiv [paper](https://arxiv.org/abs/2410.11527)|
+|"Challenging complexity with simplicity" study|Multi-step search|JCIM [paper](https://pubs.acs.org/doi/10.1021/acs.jcim.4c00432)|
```
Comment (kmaziarz) [inline README.md:53]: `"Challenging complexity with simplicity" study` could also be just `SimpRetro` if we're going for using the model/algorithm names from those papers

**Review by kmaziarz — APPROVED** [review]
LGTM!

File: README.md (hunk 1 above)
Comment (kmaziarz) [reply README.md:53]: Also, their code is under [github.com/catalystforyou/SimpRetro](https://github.com/catalystforyou/SimpRetro)

Comment (AustinT) [reply README.md]: Good catch- changed!

**Pull request merged**